agent/handlers/v2/response.go

```diff
@@ -302,9 +238,24 @@ func NewContainerResponse(
 	return resp
 }
 
+// Converts apicontainer HealthStatus type to v2 Metadata HealthStatus type
+func dockerContainerHealthToV2Health(health apicontainer.HealthStatus) *tmdsv2.HealthStatus {
+	status := health.Status.String()
+	if health.Status == apicontainerstatus.ContainerHealthUnknown {
+		// Skip sending status if it is unknown
```

Comment:
Q. did you previously send heath status `ContainerHealthUnknown`? If yes, would this be a new change introduced to v2? Also curious how would `Since`, `ExitCode` and `Output` look like when status is `ContainerHealthUnknown` : )

Reply:
That's a good question. No we do not send `Status` when container's health status is `ContainerHealthUnknown` because the `Status` field in  `apicontainer.HealthStatus` struct is defined with `json:"status,omitempty"`.

`ContainerHealthUnknown` is the zero value of `Status`'s type so `Status = ContainerHealthUnknown` is skipped when `apicontainer.HealthStatus` is marshaled to JSON.

The `TestDockerContainerHealthToV2HealthJSON` test in this PR covers this case.

Reply:
https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/container/container.go#L111-L120

Reply:
Thanks for explaining this in detail!